OsmAnd/src/net/osmand/plus/mapmarkers/MapMarkersHelper.java

```diff
@@ -1190,6 +1105,50 @@ private void removeFromGroupsList(MapMarkersGroup group) {
 		void onSyncDone();
 	}
 
+	private void runGroupSynchronization(MapMarkersGroup group) {
+		List<MapMarker> groupMarkers = new ArrayList<>(group.getMarkers());
+		if (group.getType() == ItineraryType.FAVOURITES) {
+			FavoriteGroup favGroup = app.getFavorites().getGroup(group.getName());
+			if (favGroup == null) {
+				return;
+			}
+			group.setVisible(favGroup.isVisible());
+			if (!group.isVisible() || group.isDisabled()) {
+				removeGroupActiveMarkers(group, true);
+				return;
+			}
+			List<FavouritePoint> points = new ArrayList<>(favGroup.getPoints());
+			for (FavouritePoint fp : points) {
+				addNewMarkerIfNeeded(group, groupMarkers, new LatLon(fp.getLatitude(), fp.getLongitude()), fp.getName(), fp, null);
+			}
+		} else if (group.getType() == ItineraryType.TRACK) {
+			GpxSelectionHelper gpxHelper = app.getSelectedGpxHelper();
+			File file = new File(group.getId());
+			if (!file.exists()) {
+				return;
+			}
+
+			String gpxPath = group.getId();
+			SelectedGpxFile selectedGpxFile = gpxHelper.getSelectedFileByPath(gpxPath);
+			GPXFile gpx = selectedGpxFile == null ? null : selectedGpxFile.getGpxFile();
+			group.setVisible(gpx != null || group.isVisibleUntilRestart());
+			if (gpx == null || group.isDisabled()) {
+				removeGroupActiveMarkers(group, true);
+				return;
+			}
+
+			boolean addAll = group.getWptCategories() == null || group.getWptCategories().isEmpty();
+			List<WptPt> gpxPoints = new ArrayList<>(gpx.getPoints());
+			for (WptPt pt : gpxPoints) {
+				if (addAll || group.getWptCategories().contains(pt.category)
+						|| (pt.category == null && group.getWptCategories().contains(""))) {
+					addNewMarkerIfNeeded(group, groupMarkers, new LatLon(pt.lat, pt.lon), pt.name, null, pt);
+				}
+			}
+		}
```

Comment:
add 'else if' (group.getType() == ItineraryType.MARKERS) {

and else empty block or throw exception